user.py

```diff
@@ -36,9 +55,25 @@ def _init_user_process_logging(self):
     def run_user_process(self):
         self._init_user_process_logging()
 
+        if self.warmup_q is not None:
+            self.logger.info("User %s starting warmup", self.user_id)
```

Comment:
Yeah as a second thought, I think we don't need to start the warmup from each user. Our concurrency numbers have 1x1 mapping to the number of processes we create (1 process for each user). Also, the goal is to warm up the model therefore it has to be independent of the number of users being used in the test. With the current setup, at the lower concurrency, we may not send enough warm-up requests to the model whereas at higher concurrency we may send more than necessary requests to warmup the model. We could have a separate function with fixed independent processes that are spawned at the beginning of our test that will warmup our model instead of having it dependant on the concurrency. 

Something to think about.

Reply:
Hmmm, I'm not really sure what you mean. The number of requests sent during the warmup is not dependent on the concurrency. Currently it is hardcoded to 10 requests, but this will be updated to be a configurable number of requests. All users are started up during the warmup and run a polling loop taking any prompts added to the dataset_q and using them to make a request. If there are more users than 10, then only 10 users will get a prompt and the rest will just be in a polling loop until the 10 requests are answered. If there are fewer than 10 users, some users will make multiple requests.